R/intervals_support_funs.R

```diff
@@ -0,0 +1,363 @@
+  } else if (!is.character(data$impute)) {
+    stop("The 'impute' column in the intervals data.frame must be a character column.")
+  }
+
+  # Add an index column to preserve the original order
+  index_colname <- make.unique(c("index", names(data)))[1]
+  data[[index_colname]] <- seq_len(nrow(data))
+
+  # Get all parameter column names in the data frame
+  all_param_options <- setdiff(names(PKNCA::get.interval.cols()), c("start", "end"))
+  param_cols <- intersect(names(data), all_param_options)
+
+  # If missing, define target parameters as all parameter columns with at least one TRUE.
+  if (is.null(target_params)) {
+    target_params <- param_cols
+  } else {
+    checkmate::assert_subset(target_params, choices = all_param_options, empty.ok = TRUE)
+  }
+
+  # Identify the target interval rows based on:
+  ## 1. The target groups (perfect match)
+  target_rows <- rep(TRUE, nrow(data))
+  if (!is.null(target_groups)) {
+    target_groups_data <- data[, names(target_groups), drop = FALSE]
+    target_rows <- target_rows & (
+      do.call(paste0, target_groups_data) %in% do.call(paste0, target_groups)
+    )
+  }
+  ## 2. The target parameters (at least one calculated: not-FALSE/not-NA)
+  target_params_data <- data[, target_params, drop = FALSE]
+  target_rows <- target_rows & (
+    rowSums(!is.na(replace(target_params_data, target_params_data == FALSE, NA))) > 0
+  )
+  ## 3. The target impute method is not already present and correctly positioned
+  after_vals <- lapply(strsplit(data$impute, "[ ,]+"), function(x) {
+    after_x <- which(x == target_impute)
+    if (length(after_x) == 0) return(NA)
+    if (after_x == length(x)) Inf else after_x
+  }) |> unlist()
+  target_rows <- target_rows & (after_vals != after | is.na(after_vals))
+
+  new_intervals <- data[target_rows, ]
+
+  # If no target intervals are found, nothing to change
+  if (nrow(new_intervals) == 0) {
+    warning(paste0("No intervals found with the specified target parameters,",
+                   " groups, and/or after-change needed. No changes made."))
+    return(data[, !names(data) %in% index_colname])
+
+    # If target intervals are found...
+  } else {
+    # The new imputation should not be used non-target parameters
+    new_intervals[, setdiff(param_cols, target_params)] <- FALSE
+
+    # Index the new intervals to be after the original ones
+    new_intervals[["impute"]] <- add_impute_method(new_intervals[["impute"]], target_impute, after)
+    new_intervals[[index_colname]] <- new_intervals[[index_colname]] + 0.5
+  }
+
+  # Remove the target parameters calculation from the original target intervals
+  data[target_rows, target_params] <- FALSE
+
+  # Combine the new and original intervals
+  data <- rbind(data, new_intervals)
+
+  # Filter rows where all row values for param_cols are NA or FALSE
+  param_data <- data[, param_cols, drop = FALSE]
+  rows_no_params <- rowSums(!is.na(replace(param_data, param_data == FALSE, NA))) == 0
```

Comment:
suggestion: Similar as in previous suggestion:
```suggestion
  rows_no_params <- rowSums(replace(param_data, is.na(param_data), FALSE)) == 0
```